plugins/Vibed/Vibed.cpp

```diff
@@ -52,628 +51,472 @@ extern "C"
-			this, SLOT ( noiseWaveClicked() ) );
-
-	
-	m_usrWaveBtn = new PixmapButton( this, tr( "User-defined wave" ) );
-	m_usrWaveBtn->move( 212, 109 );
-	m_usrWaveBtn->setActiveGraphic( embed::getIconPixmap(
-				"usr_wave_active" ) );
-	m_usrWaveBtn->setInactiveGraphic( embed::getIconPixmap(
-				"usr_wave_inactive" ) );
-	m_usrWaveBtn->setToolTip(
-			tr( "User-defined wave" ) );
-	connect( m_usrWaveBtn, SIGNAL ( clicked () ),
-			this, SLOT ( usrWaveClicked() ) );
-
-
-	m_smoothBtn = new PixmapButton( this, tr( "Smooth waveform" ) );
-	m_smoothBtn->move( 79, 129 );
-	m_smoothBtn->setActiveGraphic( PLUGIN_NAME::getIconPixmap(
-			"smooth_active" ) );
-	m_smoothBtn->setInactiveGraphic( PLUGIN_NAME::getIconPixmap(
-			"smooth_inactive" ) );
-	m_smoothBtn->setChecked( false );
-	m_smoothBtn->setToolTip(
-			tr( "Smooth waveform" ) );
-	connect( m_smoothBtn, SIGNAL ( clicked () ),
-			this, SLOT ( smoothClicked() ) );
-	
-	m_normalizeBtn = new PixmapButton( this, tr( "Normalize waveform" ) );
-	m_normalizeBtn->move( 96, 129 );
-	m_normalizeBtn->setActiveGraphic( PLUGIN_NAME::getIconPixmap(
-			"normalize_active" ) );
-	m_normalizeBtn->setInactiveGraphic( PLUGIN_NAME::getIconPixmap(
-			"normalize_inactive" ) );
-	m_normalizeBtn->setChecked( false );
-	m_normalizeBtn->setToolTip(
-			tr( "Normalize waveform" ) );
-
-	connect( m_normalizeBtn, SIGNAL ( clicked () ),
-			this, SLOT ( normalizeClicked() ) );
+		this);
+
+	m_harmonic->setWindowTitle(tr("Octave"));
+
+	m_stringSelector = std::make_unique<NineButtonSelector>(
+		std::array{
+			PLUGIN_NAME::getIconPixmap("button_1_on"),
+			PLUGIN_NAME::getIconPixmap("button_1_off"),
+			PLUGIN_NAME::getIconPixmap("button_2_on"),
+			PLUGIN_NAME::getIconPixmap("button_2_off"),
+			PLUGIN_NAME::getIconPixmap("button_3_on"),
+			PLUGIN_NAME::getIconPixmap("button_3_off"),
+			PLUGIN_NAME::getIconPixmap("button_4_on"),
+			PLUGIN_NAME::getIconPixmap("button_4_off"),
+			PLUGIN_NAME::getIconPixmap("button_5_on"),
+			PLUGIN_NAME::getIconPixmap("button_5_off"),
+			PLUGIN_NAME::getIconPixmap("button_6_on"),
+			PLUGIN_NAME::getIconPixmap("button_6_off"),
+			PLUGIN_NAME::getIconPixmap("button_7_on"),
+			PLUGIN_NAME::getIconPixmap("button_7_off"),
+			PLUGIN_NAME::getIconPixmap("button_8_on"),
+			PLUGIN_NAME::getIconPixmap("button_8_off"),
+			PLUGIN_NAME::getIconPixmap("button_9_on"),
+			PLUGIN_NAME::getIconPixmap("button_9_off")},
+		0,
+		21, 39,
+		this);
 
+	// String selector is not a part of the model
+	m_stringSelector->setWindowTitle(tr("String"));
+
+	connect(m_stringSelector.get(), SIGNAL(NineButtonSelection(int)), this, SLOT(showString(int)));
+
+	showString(0);
+
+	m_smoothBtn.move(79, 129);
+	m_smoothBtn.setActiveGraphic(PLUGIN_NAME::getIconPixmap("smooth_active"));
+	m_smoothBtn.setInactiveGraphic(PLUGIN_NAME::getIconPixmap("smooth_inactive"));
+	m_smoothBtn.setChecked(false);
+	m_smoothBtn.setToolTip(tr("Smooth waveform"));
+	connect(&m_smoothBtn, SIGNAL(clicked()), this, SLOT(smoothClicked()));
```

Comment:
Just a note for later enhancements: I feel we could change this to a function and just pass the variable and parameters to it.